pkg/storage/tsdb/bucketcache/caching_bucket.go

```diff
@@ -247,11 +309,14 @@ func (cb *CachingBucket) Get(ctx context.Context, name string) (io.ReadCloser, e
 	}
 
 	storeExistsCacheEntry(ctx, existsKey, true, getTime, cfg.cache, cfg.existsTTL, cfg.doesntExistTTL)
+
+	releaseSlabs = false
 	return &getReader{
 		c:         cfg.cache,
 		ctx:       ctx,
 		r:         reader,
 		buf:       new(bytes.Buffer),
+		slabs:     slabs,
```

Comment:
I don't think we need to pass the slabs at all here. Reason is that `getReader` is used on cache miss, so there will be no memory from the pool. I think in this case we can just release the pool once we exit `Get()`.

Reply:
Ah, good point.